### ISSUE TITLE
Visual Studio does not retain 'Do not use top-level statements' check.

### ISSUE THREAD

**Issue by TanvirArjel** [dotnet/aspnetcore#41631]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

First of all, thank you ASP.NET Core Team for adding an option in templates to avoid top-level statement. 

However, we have noticed that  Visual Studio is not retaining the `Do not use top-level statement` check.

First Time:

![Capture1](https://user-images.githubusercontent.com/14342773/167901057-0a69aa8b-a6e7-4045-887f-b8fbd331711d.PNG)

Second Time:

![Capture2](https://user-images.githubusercontent.com/14342773/167901147-1f9eac42-ee40-43d4-9478-88c5c125df7f.PNG)

Note: All other checks are retaining their last choice.

### Expected Behavior

Check should retain the last choice.

### .NET Version

6.0.300 and 7.0.100-preview.4.22252.9

### Anything else?

Visual Studio Version: 2022 17.3 preview 1.

**Comment by TanayParikh** [comment]
cc/ @DamianEdwards as I believe this was recently added.

**Comment by TanvirArjel** [comment]
@mkArtakMSFT Is it not part of templating? However, please redirect to the appropriate channel.

**Comment by DamianEdwards** [comment]
@phenning

**Comment by DamianEdwards** [comment]
Details for the change we need to make to the `ide.host.json` for each web project template, including the console template, in 7.0:

```json
{
  "id": "UseProgramMain",
  "isVisible": true,
  "PersistenceScope": "Shared",
  "PersistenceScopeName": "Microsoft"
}
```

@phenning is going to ensure that qnce we make this change in 7.0, any customer with both 6.x and 7.0 installed will see this option become sticky for all versions of the template available, i.e. it will become sticky in 6.x templates too.

/Cc @vlada-shubina

**Comment by TanvirArjel** [comment]
Hello @DamianEdwards , Here is another issue: https://github.com/dotnet/aspnetcore/issues/41636

**Comment by DamianEdwards** [comment]
Thanks, that's a known issue that we're currently working on getting a fix in to 7.0 and 6.0 for (#41491).

**Comment by TanvirArjel** [comment]
@DamianEdwards I have checked with **.NET SDK 6.0.7 and 7.0.0 preview6**. The issue has not been fixed.

**Comment by DamianEdwards** [comment]
Yeah this fix is in the 7.0 SDK and thus will only apply to 6.0 projects when 7.0.100-preview.6 is installed. VS 17.4 will default to including .NET 7 SDK at which point this will be fixed for 6.0 projects too.
